src/playsim/p_map.cpp

```diff
@@ -4312,17 +4312,39 @@ DAngle P_AimLineAttack(AActor *t1, DAngle angle, double distance, FTranslatedLin
 
 	aim_t aim;
 
+	DVector3 pos = t1->Pos();
+	DAngle attackPitch = t1->Angles.Pitch;
+	DAngle attackAngle = angle;
+	if (t1->player != nullptr && t1->player->mo->OverrideAttackPosDir)
+    {
+	    pos = t1->player->mo->AttackPos;
+
+        //
+        //  FIX FOR FLAT PROJECTILE SPREAD FROM SHOTGUNS AND OTHER MULTI-PROJECTILE WEAPONS
+        //
+        //Bit of a hack as it makes the pitch a little off, but in order for weapon projectile spread
+        //to work correctly, we can't fix the pitch to that of the controller otherwise random spread
+        //turns into a flat line of projectiles. Using the difference between the previous known angles and
+        //the current angles (which were potenitally modified by A_FireProjectile) gives us the "random"
+        //difference applied to the pitch by the script
+        //
+        //A VR player is unlikely to be whipping their head up and down so fast as to make a meaningful
+        //difference to the pitch between two frames, so this is 'good enough' for now.
+        attackPitch = t1->player->mo->AttackPitch + (t1->PrevAngles.Pitch - t1->Angles.Pitch);
+        attackAngle = t1->player->mo->AttackAngle + (angle - t1->Angles.Yaw);
```

Comment:
In this fork AActor is missing AttackPitch and AttackAngle, do you think I can simply delete it ? I have tested without themm and it seems working fine, at least with SSG

Reply:
Yes, you can definitely leave that specific bit of the change out without any ill effects.